### PR TITLE
Add Notifications Drawer and Toast Notifications List

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
This PR replaces the use of inline notifications and the Angular Patternfly Notifications service with a Notifications Drawer, Toast List, and EventNotifications service.

The inline messages are now shown as Toasts in the upper right section of the window and are then also available in the Notifications Drawer until removed by the user. The notifications shown in the drawer are for the current session only, there is no saving/retrieval of past notifications or notifications from other sessions.

Toasts:
image

Notifications Drawer:
image

@serenamarie125 @beanh66 @chriskacerguis @dtaylor113

### CODE REVIEW

**Comment by dtaylor113** [comment]
Please merge this after #131 is merged!  Some of the blueprint files were reverted in #131, so are outdated in this PR.  -thanks

File: client/app/components/notifications/notification-body.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <span ng-if="notification.type" class="pull-left {{customScope.getNotficationStatusIconClass(notification)}}" ng-click="customScope.markRead(notification, notificationGroup)"></span>
+      <span class="drawer-pf-notification-message notification-message"
+            ng-click="customScope.markRead(notification, notificationGroup)"
+            tooltip-append-to-body="true" tooltip-popup-delay="500" tooltip-placement="bottom" tooltip="{{notification.message | translate}}">
+            {{notification.message | translate}}
+      </span>
+    </div>
+    <div class="col-md-4 col-sm-3" ng-if="notificationGroup.notificationType == 'task'">
+      <div class="drawer-pf-notification-info expanded-info" ng-click="customScope.markRead(notification, notificationGroup)">
+        <span class="info-title">Started: </span>
```
Comment (AllenBW) [inline client/app/components/notifications/notification-body.html]: no translate attribute on this span?

File: client/app/components/notifications/notification-body.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      </div>
+      <div ng-if="notification.inProgress" class="progress">
+        <div class="progress-bar progress-bar-info"
+             ng-style="{width: notification.data.percentComplete + '%'}" tooltip="{{notification.data.percentComplete}} + {{'% Complete' | translate}}">
+        </div>
+      </div>
+    </div>
+    <div class="col-md-4 col-sm-3" ng-if="notificationGroup.notificationType == 'task'">
+      <div class="drawer-pf-notification-info" ng-click="customScope.markRead(notification, notificationGroup)">
+        <span class="info-title">Completed: </span>
```
Comment (AllenBW) [inline client/app/components/notifications/notification-body.html]: Also this one, translate attribute?

**Comment by jeff-phillips-18** [comment]
Corresponding Jira Stories:

Design: https://patternfly.atlassian.net/browse/MGMTUX-179
Initial Dev: https://patternfly.atlassian.net/browse/MGMTUX-180
Final Dev: https://patternfly.atlassian.net/browse/CFUX-92

**Comment by jeff-phillips-18** [comment]
@AllenBW I looked at replacing toastr as well but I agree it's better left to another PR since the fields are slightly different and I didn't want to get into it all at once.

**Comment by dtaylor113** [comment]
Hi Jeff, if you have to update this PR, could you remove from client/app/states/designer/blueprints/list/list.state.js:
```
/* This notification 'splice' code doesn't work.  Splice needs a third argument, the items to splice in
     * Not sure what this code is trying to accomplish, but it exists in login, request list, & services list
    if (angular.isDefined($rootScope.notifications) && $rootScope.notifications.data.length > 0) {
      $rootScope.notifications.data.splice(0, $rootScope.notifications.data.length);
    }
*/
```
If it is no longer relevant.  If not, I'll remove it in my next PR.  -thanks

**Comment by dtaylor113** [comment]
LGTM :+1:

**Comment by serenamarie125** [comment]
From a UX perspective, this is good to go!

File: client/app/components/notifications/notification-body.html (hunk 2 above)
Comment (jeff-phillips-18) [inline client/app/components/notifications/notification-body.html]: Do we prefer the translate attribute or the {{"some string" | translate}} or does it matter?

Also, is there a date format translator? That is something better than
  {{notification.timeStamp | date:'MM/dd/yyyy'}}
or
  {{notification.timeStamp | date:'h:mm:ss a'}}

That will format based on locale?

**Comment by dtaylor113** [comment]
For reviewers, try out the new github review process.  Click on "Files Changed" above, then on the upper right "Review Changes" green button.

File: client/app/components/notifications/notification-body.html (hunk 2 above)
Comment (AllenBW) [inline client/app/components/notifications/notification-body.html]: This is a darn good question, I don't know if it matters, i like the idea of only the strings though, looping in @himdel for some opinions here..

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 looks like this needs to be rebased

**Comment by chriskacerguis** [comment]
👍

File: client/app/components/notifications/notification-body.html (hunk 2 above)
Comment (himdel) [inline client/app/components/notifications/notification-body.html]: I'd say prefer translate attribute *if* you can sanely use it, but go for `{{""|translate}}` if you'd have to create elements just to be able to do the attribute :).

As for l10n, we simply currently don't support it. Would be nice though.. And IIRC we do have `moment` in there, with locale support, so maybe we could make a moment filter and set the appropriate moment locale when changing the active language .. and get at least a rudimentary l10n that way... OTOH, I'd say it's better to be consistently wrong than being inconsistently right, so.. if our date picker can't handle different formats reasonably, it'd be more suprising to have one format in edit and one in view than to force the US way in both..

File: client/app/states/designer/blueprints/list/list.state.js [hunk 3]
```diff
@@ -86,13 +86,6 @@
       visibilityNames.push(item.name);
     }
 
-    /* This notification 'splice' code doesn't work.  Splice needs a third argument, the items to splice in
```
Comment (himdel) [inline client/app/states/designer/blueprints/list/list.state.js]: @jeff-phillips-18 Jeff, this code does work, splice doesn't need more than 2 arguments... it clears the array without destroying any references :).

You can think of it as $rootScope.notifications.data = []; except you can be sure that if you've done foo = $rootScope.notifications.data before, then even foo will be empty (and still point to the same array).

**Pull request merged**